R/drug_parameters.R

```diff
@@ -1,44 +1,95 @@
+#' @details Default parameters, from L to R, are: drug_efficacy: 0.9, drug_rel_c: 0.32, drug_prophylaxis_shape: 4.3, drug_prophylaxis_scale: 38.1
+#' @export
+SP_AQ_params_falciparum <- c(0.9, 0.32, 4.3, 38.1)
+
 #' @title Preset parameters for the SP-AQ drug
 #' @details Use a vector of preset parameters for the SP-AQ drug (sulphadoxine-pyrimethamine and amodiaquine)
 #' @details Default parameters, from L to R, are: drug_efficacy: 0.9, drug_rel_c: 0.32, drug_prophylaxis_shape: 4.3, drug_prophylaxis_scale: 38.1
 #' @export
 SP_AQ_params <- c(0.9, 0.32, 4.3, 38.1)
 
+#' @title Preset parameters for the CQ drug acting on P. vivax
+#' @description Efficacy from SI of Nekkab et al., DOI: 10.1371/journal.pmed.1003535 (2021),
+#' shape and scale consistent with a longer prophylaxis time: 28 days, decreasing gradually
+#' @details Use a vector of preset parameters for the CQ drug (chloroquine) acting on P. vivax
+#' @details Default parameters, from L to R, are: drug_efficacy: 0.899, drug_rel_c: 0.5, drug_prophylaxis_shape: 20, drug_prophylaxis_scale: 5
+#' @export
+CQ_params_vivax <- c(0.899, 0.5, 5, 20)
+
+#' @title Preset parameters for the CQ-PQ drug acting on P. vivax
+#' @description Efficacy from SI of Nekkab et al., DOI: 10.1371/journal.pmed.1003535 (2021),
+#' BS shape and scale consistent with a longer prophylaxis time: 28 days, decreasing gradually,
+#' LS shape are scale consistent with a 7 day PQ treatment, with rapid decay.
+#' @details Use a vector of preset parameters for the CQ-PQ drug (chloroquine and primaquine) acting on P. vivax
+#' @details Default parameters, from L to R, are: drug_efficacy: 0.948, drug_rel_c: 0.5, drug_prophylaxis_shape: 5, drug_prophylaxis_scale: 20,
+#' drug_hypnozoite_efficacy: 0.713, drug_hypnozoite_prophylaxis_shape: 10, drug_hypnozoite_prophylaxis_scale: 5.5
+#' @export
+CQ_PQ_params_vivax <- c(0.948, 0.5, 5, 20, 0.713, 10, 5.5)
+
+#' @title Preset parameters for the CQ-TQ drug acting on P. vivax
+#' @description Efficacy from SI of Nekkab et al., DOI: 10.1371/journal.pmed.1003535 (2021),
+#' BS shape and scale consistent with a longer prophylaxis time: 28 days, decreasing gradually,
+#' LS shape are scale consistent with a single TQ treatment, with longer prophylaxis: 45 days.
+#' @details Use a vector of preset parameters for the CQ-TQ drug (chloroquine and tafenoquine) acting on P. vivax
+#' @details Default parameters, from L to R, are: drug_efficacy: 1, drug_rel_c: 0.5, drug_prophylaxis_shape: 5, drug_prophylaxis_scale: 20,
+#' drug_hypnozoite_efficacy: 0.713, drug_hypnozoite_prophylaxis_shape: 5.5, drug_hypnozoite_prophylaxis_scale: 30
+CQ_TQ_params_vivax <- c(1, 0.5, 5, 20, 0.713, 5, 30)
+
 #' @title Parameterise drugs to use in the model
 #'
 #' @param parameters the model parameters
 #' @param drugs a list of drug parameters, can be set using presets
 #' @export
 set_drugs <- function(parameters, drugs) {
```

Comment:
Is there any validation we could do here to validate if parameters$parasite == vivax that the correct drugs are used? Or do all drugs make sense for all parasites?

Reply:
I spoke with Pete about this a while ago, and we came to a place of hoping that people would use their good sense to choose which parameter sets to apply for each species. While drug parameters are parasite-specific, there might be circumstances where someone would want to see what happens if the same drug is applied to both species. I've added a few warnings to flag where people might do something strange while maintaining flexibility. I could change some of the warnings to errors (e.g., the length of the parameter inputs) if we think that's important.